utils/resultstable.go

```diff
@@ -930,15 +930,14 @@ func convertCves(cves []services.Cve) []formats.CveRow {
 
 // If at least one cve is applicable - final value is applicable
 // Else if at least one cve is undetermined - final value is undetermined
+// Else if all cves are not covered -> final value is not covered
 // Else (case when all cves aren't applicable) -> final value is not applicable
 func getApplicableCveStatus(entitledForJas bool, applicabilityScanResults []*sarif.Run, cves []formats.CveRow) ApplicabilityStatus {
-	if !entitledForJas || len(applicabilityScanResults) == 0 {
+	if !entitledForJas || len(applicabilityScanResults) == 0 || len(cves) == 0 {
 		return NotScanned
 	}
-	if len(cves) == 0 {
-		return ApplicabilityUndetermined
-	}
 	foundUndetermined := false
```

Comment:
@attiasas I didn't use here the new function I created `setApplicabilityStatusFromRule` because the new function expect to get an array of statuses and here we have an array of cves.
I'm not sure what's better - loop over the cves to get the statuses (and covert the type of the status to string) in order to use the new function, or keep this duplicated logic.

Reply:
To avoid duplication, you can do the following:
avoiding duplications is better. loop over the Cves and collect the statuses.